src/model/CoilCoolingDXCurveFitSpeed.cpp

```diff
@@ -0,0 +1,689 @@
+    bool result = setPointer(OS_Coil_Cooling_DX_CurveFit_SpeedFields::WasteHeatModifierFunctionofTemperatureCurve, curve.handle());
+    OS_ASSERT(result);
+    return result;
+  }
+
+  bool CoilCoolingDXCurveFitSpeed_Impl::setRatedWasteHeatFractionofPowerInput(double ratedWasteHeatFractionofPowerInput) {
+    bool result = setDouble(OS_Coil_Cooling_DX_CurveFit_SpeedFields::RatedWasteHeatFractionofPowerInput, ratedWasteHeatFractionofPowerInput);
+    return result;
+  }
+
+  bool CoilCoolingDXCurveFitSpeed_Impl::setSensibleHeatRatioModifierFunctionofTemperatureCurve(const Curve& curve) {
+    bool result = setPointer(OS_Coil_Cooling_DX_CurveFit_SpeedFields::SensibleHeatRatioModifierFunctionofTemperatureCurve, curve.handle());
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void CoilCoolingDXCurveFitSpeed_Impl::resetSensibleHeatRatioModifierFunctionofTemperatureCurve() {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_SpeedFields::SensibleHeatRatioModifierFunctionofTemperatureCurve, "");
+    OS_ASSERT(result);
+  }
+
+  bool CoilCoolingDXCurveFitSpeed_Impl::setSensibleHeatRatioModifierFunctionofFlowFractionCurve(const Curve& curve) {
+    bool result = setPointer(OS_Coil_Cooling_DX_CurveFit_SpeedFields::SensibleHeatRatioModifierFunctionofFlowFractionCurve, curve.handle());
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void CoilCoolingDXCurveFitSpeed_Impl::resetSensibleHeatRatioModifierFunctionofFlowFractionCurve() {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_SpeedFields::SensibleHeatRatioModifierFunctionofFlowFractionCurve, "");
+    OS_ASSERT(result);
+  }
+
+  boost::optional<double> CoilCoolingDXCurveFitSpeed_Impl::autosizedGrossSensibleHeatRatio() const {
+    return getAutosizedValue("TODO_CHECK_SQL Gross Sensible Heat Ratio", "");
+  }
+
+  void CoilCoolingDXCurveFitSpeed_Impl::autosize() {
+    autosizeGrossSensibleHeatRatio();
+  }
+
+  void CoilCoolingDXCurveFitSpeed_Impl::applySizingValues() {
+    boost::optional<double> val;
+    val = autosizedGrossSensibleHeatRatio();
+    if (val) {
+      setGrossSensibleHeatRatio(val.get());
+    }
+
+  }
+
+  ModelObject CoilCoolingDXCurveFitSpeed_Impl::clone(Model model) const {
+    auto t_clone = ModelObject_Impl::clone(model).cast<CoilCoolingDXCurveFitSpeed>();
+
+    return t_clone;
+  }
+
+  std::vector<ModelObject> CoilCoolingDXCurveFitSpeed_Impl::children() const {
+    std::vector<ModelObject> result;
+
+    result.push_back(totalCoolingCapacityModifierFunctionofTemperatureCurve());
+    result.push_back(totalCoolingCapacityModifierFunctionofAirFlowFractionCurve());
+    result.push_back(energyInputRatioModifierFunctionofTemperatureCurve());
+    result.push_back(energyInputRatioModifierFunctionofAirFlowFractionCurve());
+    result.push_back(partLoadFractionCorrelationCurve());
+    result.push_back(wasteHeatModifierFunctionofTemperatureCurve());
```

Comment:
This line should probably become the following after it's made optional:

```c++
if (auto _c = wasteHeatModifierFunctionofTemperatureCurve()) {
  result.push_back(_c.get());
}
```

I think you are missing two additional optional curves too:

```
  boost::optional<Curve> sensibleHeatRatioModifierFunctionofTemperatureCurve() const;

  boost::optional<Curve> sensibleHeatRatioModifierFunctionofFlowFractionCurve() const;
```